eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -24,43 +24,59 @@
 package org.eolang.maven;
 
 import com.jcabi.log.Logger;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import org.cactoos.Input;
 import org.cactoos.io.InputOf;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /**
  * The simple HTTP Objectionary server.
  *
  * @since 0.1
  */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals",
+    "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"})
 public final class OyRemote implements Objectionary {
 
     /**
      * The address template.
      */
     private final String template;
 
-    /**
-     * Ctor.
-     */
-    public OyRemote() {
-        this("master");
-    }
-
     /**
      * Ctor.
      * @param hash The GitHub hash
-     * @todo #490:30m Resolve abbreviated hash to a proper hash.
-     *  In order to avoid collisions resolve hash
-     *  (or branch) to a complete sha-256 hash of the commit.
-     *  Use only sha-256 hashes as a caching criteria.
+     * @throws JSONException if fails
+     * @throws IOException if fails
      */
-    public OyRemote(final String hash) {
+    public OyRemote(final String hash) throws IOException, JSONException {
+        final int limit = 2;
+        int tries = 0;
+        String sha;
+        while (true) {
+            try {
+                sha = this.getSha(hash);
+                break;
+            } catch (final IOException | JSONException exception) {
+                tries = tries + 1;
+                if (tries == limit) {
+                    Logger.info(this, "Couldn't get commit SHA. It will be set as \"master\"");
```

Comment:
@yasamprom this message is misleading. The exception is thrown, nothing will be "set". I suggest we just throw an exception and that's it. The logging here is not necessary.

Reply:
@yegor256 Done